tools/chplcheck/src/rules.py

```diff
@@ -26,6 +26,40 @@
 from fixits import Fixit, Edit
 from rule_types import BasicRuleResult, AdvancedRuleResult
 
+def variables(node: AstNode):
+    if isinstance(node, Variable):
+        if node.name() != "_":
+            yield node
+    elif isinstance(node, TupleDecl):
+        for child in node:
+            yield from variables(child)
+
+def file_lines_for_node(context: Context, node: AstNode):
+     return context.get_file_text(node.location().path()).split("\n")
+
+def fixit_remove_unused_node(node: AstNode, lines: Optional[List[str]] = None, context: Optional[Context] = None):
```

Comment:
maybe a doc comment here describing that `lines` or `context` must be supplied, and what `node` types this works on (iterand of a IndexableLoop and TupleDecl)